src/Illuminate/Console/Scheduling/Schedule.php

```diff
@@ -49,6 +50,19 @@ public function call($callback, array $parameters = [])
     }
 
     /**
+     * Add a new queued job callback event to the schedule.
+     *
+     * @param  \Illuminate\Contracts\Queue\ShouldQueue  $job
+     * @return \Illuminate\Console\Scheduling\Event
+     */
+    public function job(ShouldQueue $job)
```

Comment:
Name is used solely for the cache mutex name. I don't see a case where you'd want to set that as anything other than the full class name of the Job.